### PR TITLE
Fix checklist column display on hierarchical CPTs list table

### PR DESCRIPTION
**Pull request by mattheu**
BUG: For custom post type, "publishing checklist" column header is showing on post list table but not the evaluated results.

Whats the problem? got to love the consistency of WP. 

the action `manage_posts_columns` is fired for all post types that are NOT `pages`.
the action `manage_posts_custom_column ` is fired for all post types that are NOT heriarchical

Notice that these are not the same ;) My post type is heirarchical but not a page, and only gets the latter.

Solution here?

Would be nice to add the action by post type, for all post types with tasks attached. This is OK as long as we hook in later (after tasks registered) but before rendering the post list table. 

What do you think of this solution?

### CODE REVIEW

**Comment by danielbachhuber** [comment]
> What do you think of this solution?

Looks reasonable to me. @goldenapples ?

Can you merge master into this branch?

File: publishing-checklist.php [hunk 1]
```diff
@@ -35,10 +35,31 @@ public static function get_instance() {
...
+			foreach ( $post_types as $post_type ) {
+				add_action( "manage_{$post_type}_posts_custom_column", array( $this, 'action_manage_posts_custom_column' ), 10, 2 );
+				add_filter( "manage_{$post_type}_posts_columns", array( $this, 'filter_manage_posts_columns' ), 99 );
+			}
+
+		} );
+
+	}
+
+	private function get_supported_post_types() {
```
Comment (goldenapples) [inline publishing-checklist.php]: Any reason why this function is left in? Were you intending on moving the loop through post types to this helper method?

**Comment by goldenapples** [comment]
> got to love the consistency of WP.

Wow, yeah, that's a weird bug.

This seems like a reasonable enough solution to me, other than my question above about the empty method you're adding.

I also wonder if it doesn't make sense to just check `get_current_screen()['post_type']` and only add actions for that post type. But its all the same result.

File: publishing-checklist.php (hunk 1 above)
Comment (mattheu) [inline publishing-checklist.php]: Oh yeah - I think I should just remove

**Comment by mattheu** [comment]
> I also wonder if it doesn't make sense to just check get_current_screen()['post_type'] and only add actions for that post type. But its all the same result.

Yeah I thought about this. Thats probably a neater solution, unless anyone is trying to show multiple post types in the same list table? What do you reckon?

**Comment by danielbachhuber** [comment]
> Thats probably a neater solution, unless anyone is trying to show multiple post types in the same list table?

Doesn't work, and you shouldn't try it :) https://github.com/fusioneng/Unified-Post-Types

**Comment by goldenapples** [comment]
> unless anyone is trying to show multiple post types in the same list table

Ah, I guess that's possible, and a good case to allow support for, since its easy enough. Might as well leave it as is in that case.

:+1: :shipit:

**Comment by goldenapples** [comment]
> Doesn't work, and you shouldn't try it 

:ok: good to know, haha.

**Pull request merged**